### PR TITLE
Supplement the code differences of Xiaomi vela's CAN modules.

### PR DESCRIPTION
**Pull request by OceanfromXiaomi**
## Summary
Supplement the code differences of Xiaomi vela's CAN modules in recent years.

### CODE REVIEW

File: drivers/can/can.c [hunk 1]
```diff
@@ -89,6 +89,12 @@
  * Private Function Prototypes
  ****************************************************************************/
 
+/* CAN helpers */
+
+static uint8_t        can_dlc2bytes(uint8_t dlc);
+#if 0 /* Not used */
+static uint8_t        can_bytes2dlc(uint8_t nbytes);
+#endif
```
Comment (acassis) [inline drivers/can/can.c:97]: Why is it not used? If it is not used, why to keep it in the source code?

**Review by anchao — CHANGES_REQUESTED** [review]

**Comment by raiden00pl** [comment]
please add information about what was changed in this commit and why, so that someone viewing the git history doesn't have to analyze the change and guess what the author's intention was

**Comment by michallenc** [comment]
I am puzzled by this... it seems to partially be the revert of 816b1b28a and e16ee4dba (and possibly some other commits), but without any obvious reason. We NEED `can_bytes2dlc` and `can_dlc2bytes` functions to be in public API because application has to fill CAN frame length directly as dlc instead of bytes (and dlc != bytes for FD frames). So we have to provide some API to do this. Also we need some way to obtain and set data bittiming for FD capable controllers.

Was there any reason to revert these changes? Because this seems like your branch is against older NuttX instead of upstream... try rebase. Anyway, in my opinion, some of these changes are a significant setback for current CAN API and I am strongly against merging it in the current state.

File: drivers/can/can.c (hunk 1 above)
Comment (OceanfromXiaomi) [reply drivers/can/can.c:97]: OK, these codes are changed by my teacher, I will tell him submit these codes to nuttx community and describe the reason about each topic.

File: include/nuttx/can/can.h [hunk 2]
```diff
@@ -946,46 +931,6 @@ int can_txdone(FAR struct can_dev_s *dev);
...
- *
- * Input Parameters:
- *   nbytes - the byte count to convert to a DLC value
- *
- * Returned Value:
- *   The encoded DLC value corresponding to at least that number of bytes.
- *
- ****************************************************************************/
-
-uint8_t can_bytes2dlc(uint8_t nbytes);
```
[the remark this answers is not in the document]
Comment (OceanfromXiaomi) [reply include/nuttx/can/can.h:967]: OK, these codes are changed by my teacher, I will tell him submit these codes to nuttx community and describe the reason about each topic.

**Comment by OceanfromXiaomi** [comment]
> please add information about what was changed in this commit and why, so that someone viewing the git history doesn't have to analyze the change and guess what the author's intention was

OK, these codes are changed by my teacher, I will tell him submit these codes to nuttx community and describe the reason about each topic.

**Comment by OceanfromXiaomi** [comment]
> I am puzzled by this... it seems to partially be the revert of [816b1b2](https://github.com/apache/nuttx/commit/816b1b28a0748861b9eb7abc55d1c8bb94e97fde) and [e16ee4d](https://github.com/apache/nuttx/commit/e16ee4dba3a403d6c78e54a266d93912861b5750) (and possibly some other commits), but without any obvious reason. We NEED `can_bytes2dlc` and `can_dlc2bytes` functions to be in public API because application has to fill CAN frame length directly as dlc instead of bytes (and dlc != bytes for FD frames). So we have to provide some API to do this. Also we need some way to obtain and set data bittiming for FD capable controllers.
> 
> Was there any reason to revert these changes? Because this seems like your branch is against older NuttX instead of upstream... try rebase. Anyway, in my opinion, some of these changes are a significant setback for current CAN API and I am strongly against merging it in the current state.

OK, these codes are changed by my teacher, I will tell him submit these codes to nuttx community and describe the reason about each topic.

**Pull request closed without merging**